data-prepper-api/src/main/java/org/opensearch/dataprepper/model/configuration/DataPrepperVersion.java

```diff
@@ -0,0 +1,101 @@
+package org.opensearch.dataprepper.model.configuration;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class DataPrepperVersion {
```

Comment:
Did we decide on whether this was specifying the underlying version of DataPrepper or the config version?

Reply:
Yeah actually after thinking about this some more. I don't think it makes sense to have a DataPrepper version in the YAML the version is going to be whatever version the binary you are using is. The versioning should be the versioning of the config. I.e. the first check in the config file should be for the version. If the version is specified, then use the appropriate java class to deserialize the config into.

Reply:
Happy to change the internal class name. Right now every major and minor version of Data Prepper signifies a change in pipeline configuration. We have been using one version for the application and configuration. Adding support for configuration allows will give us the option to break these apart in the future. The validation checks are super simple at the moment. 

RE: deserializing different config objects based on the version - I am not sure we want to do that at this time. We don't have a need to support nor different config versions right now. I am not sure its in our best interest to support this either. That will become very difficult to maintain.

Reply:
So whenever we do a version bump on Data Prepper, this file will be updated?

Reply:
Yes, that is the plan: https://github.com/opensearch-project/data-prepper/pull/2292#discussion_r1112321785